app/validation/validate_schema_text.py

```diff
@@ -0,0 +1,64 @@
+import re
+from jsonpointer import resolve_pointer
+from eq_translations.utils import find_pointers_to
+from eq_translations.survey_schema import SurveySchema
+
+
+def get_pointer_keys():
+    pointer_keys = [
+        'show_guidance',
+        'hide_guidance',
+        'description',
+        'legal_basis',
+        'playback',
+        'item_title',
+        'label',
+        'add_link_text',
+        'empty_list_text',
+        'text',
+        'title',
+    ]
+
+    return pointer_keys
+
+
+def get_pointer_contents(schema, keys):
+    contents = []
+    for key in keys:
+        schema_elements = find_pointers_to(schema, key)
+        for element in schema_elements:
+            pointer_contents = resolve_pointer(schema, element)
+            contents.append(pointer_contents)
+
+    return contents
+
+
+def get_list_pointer_contents(schema, schema_elements):
+    contents = []
+    for element in schema_elements:
+        pointer_contents = resolve_pointer(schema, element)
+        contents.append(pointer_contents)
+    return contents
+
+
+def search_dumb_quotes(text):
+    return re.search(r"'", text)
+
+
+def validate_json_schema_text(json_schema, schema_path):
+    error_message = 'Found dumb quotes in schema text'
+    schema_contents = get_pointer_contents(json_schema, get_pointer_keys())
+
+    schema_object = SurveySchema()
+    schema_object.load(schema_path)
+
+    list_pointers = schema_object.get_list_pointers()
+    list_contents = get_list_pointer_contents(json_schema, list_pointers)
+
+    for list_text in list_contents:
+        if search_dumb_quotes(list_text):
+            return error_message
+
+    for schema_text in schema_contents:
+        if search_dumb_quotes(schema_text):
+            return error_message
```

Comment:
- We should be generating an error for each string that contains a dumb quote. 
- We should create a single consolidated list of pointers, and then iterate over that.

Something like:

```python
pointers = []
errors = []

for key in get_pointer_keys():
    pointers.extend(find_pointers_to(json_schema, key))

schema_object = SurveySchema()
schema_object.load(schema_path)
pointers.extend(schema_object.get_list_pointers())

for pointer in pointers:
    schema_text = resolve_pointer(schema, pointer)
    if search_dumb_quotes(list_text):
        errors.append(f'Found dumb quote(s) in schema text at [{pointer}]')
```

Reply:
Refactored this